include/aspect/material_model/rheology/friction_options.h

```diff
@@ -0,0 +1,198 @@
+           * MaterialModelOutputs object that is handed over.
+           */
+          void compute_theta_reaction_terms(const int q,
+                                            const MaterialModel::MaterialModelInputs<dim> &in,
+                                            const std::vector<double> &volume_fractions,
+                                            const double min_strain_rate,
+                                            const double ref_strain_rate,
+                                            bool use_elasticity,
+                                            bool use_reference_strainrate,
+                                            const std::vector<double> &elastic_shear_moduli,
+                                            const double dte,
+                                            MaterialModel::MaterialModelOutputs<dim> &out) const;
+
+          /**
+           * A function that returns the selected type of friction dependence.
+           */
+          FrictionDependenceMechanism
+          get_friction_dependence_mechanism () const;
+
+          /**
+           * A function that returns if the state variable theta is used.
+           */
+          bool get_theta_in_use() const;
+
+        private:
+
+          FrictionDependenceMechanism friction_dependence_mechanism;
+
+
+          /*
+           * Objects for computing plastic stresses, viscosities, and additional outputs
+           */
+          Rheology::DruckerPrager<dim> drucker_prager_plasticity;
+          /**
+          * Input parameters for the drucker prager plasticity.
+          */
+          Rheology::DruckerPragerParameters drucker_prager_parameters;
+
+          /**
+          * Object for computing viscoelastic viscosities and stresses.
+          */
+          Rheology::Elasticity<dim> elastic_rheology;
+
+          Rheology::StrainDependent<dim> strain_rheology;
+
+          /**
+          * dynamic friction input parameters
+          */
+          std::vector<double> dynamic_angles_of_internal_friction;
+          double dynamic_characteristic_strain_rate;
+          double dynamic_friction_smoothness_exponent;
+
+          /**
+          * rate and state friction input parameters
+          */
+          std::vector<double> effective_friction_factor;
+          double critical_slip_distance;
+          std::vector<double> steady_state_strain_rate;
+
+          /**
+           * Parsed functions that specify a and b depth-dependence when using the Function
+           * method.
+           */
+          std::unique_ptr<Functions::ParsedFunction<dim> > rate_and_state_parameter_a_function;
+          std::unique_ptr<Functions::ParsedFunction<dim> > rate_and_state_parameter_b_function;
+
+          /**
+           * Function to calculate depth-dependent a and b values for state dependent friction.
+           */
+          std::pair<double,double>
+          calculate_depth_dependent_a_and_b(const Point<dim> &position, const int j) const;
```

Comment:
The usual order is class functions and then parameters

Reply:
ok, I thought it would be convenient to have everything concerning a and b at one place in the script. What do you think?

Reply:
For me, this order of functions and then parameters takes precedence. It helps when you're reading code when your expectations are met regarding code structure. That said, looking at some ASPECT header files, this order is not always kept. So it's up to you.